sycl/include/sycl/ext/intel/math/imf_bf16.hpp

```diff
@@ -0,0 +1,205 @@
+//==-------------------- imf_bf16.hpp - bfloat16 utils ---------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// C++ APIs for bfloat16 util functions.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+#include <sycl/ext/oneapi/experimental/bfloat16.hpp>
+#include <type_traits>
+
+using sycl_bfloat16 = sycl::ext::oneapi::experimental::bfloat16;
+using _iml_bfloat16_internal = uint16_t;
+
+extern "C" {
+float __imf_bfloat162float(_iml_bfloat16_internal);
+_iml_bfloat16_internal __imf_float2bfloat16(float);
+_iml_bfloat16_internal __imf_float2bfloat16_rd(float);
+_iml_bfloat16_internal __imf_float2bfloat16_rn(float);
+_iml_bfloat16_internal __imf_float2bfloat16_ru(float);
+_iml_bfloat16_internal __imf_float2bfloat16_rz(float);
+};
+
+namespace sycl {
+__SYCL_INLINE_VER_NAMESPACE(_V1) {
+namespace ext {
+namespace intel {
+namespace math {
+
+// Need to ensure that sycl bfloat16 defined in bfloat16.hpp is compatible
+// with uint16_t in layout.
+#if __cplusplus >= 201703L
+static_assert(sizeof(sycl_bfloat16) == sizeof(_iml_bfloat16_internal),
+              "sycl bfloat16 is not compatible with _iml_bfloat16_internal.");
```

Comment:
Why do we need a distinct alias at all?

Reply:
Hi, @aelovikov-intel 
What distinct alias do you refer to?
Thanks very much.

Reply:
Why can't we use the same type in all places? Why do we need both `sycl_bfloat16` and `_iml_bfloat16_internal`?

Reply:
Hi, @aelovikov-intel 
The functions defined in sycl::ext::intel::math:: namespace such as hge, hgt... are c++ wrappers for c functions provided in libdevice. All these c++ functions will call corresponding c functions and sycl bfloat16 users will only work with these c++ functions. sycl bfloat16 is a c++ class defined in https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/oneapi/bfloat16.hpp and its current implementation is based on "uint16_t" type. So, our c++ functions such as hge, hgt... can only accept user-visible sycl bfloat16. However, the c functions implemented in libdevice can't use c++ sycl bfloat16 type, they can only accept native C types , so we need to use native C uint16_t type for them. In the future, we may use native bfloat16 type instead emulation based on uint16_t.
Thanks very much.